posthog-core/src/index.ts

```diff
@@ -1148,7 +1150,7 @@ export abstract class PostHogCore extends PostHogCoreStateless {
   private async decideAsync(sendAnonDistinctId: boolean = true): Promise<PostHogDecideResponse | undefined> {
     await this._initPromise
     if (this._decideResponsePromise) {
-      return this._decideResponsePromise
+      await this._decideResponsePromise
```

Comment:
if reload flags were called multiple times (eg reset, reload, identify, etc), the requests should be executed one after the other with the new values (eg new distinct id, etc) rather than returning the current  request maybe with the wrong request values